File: app/src/test/java/uk/co/ianfield/devstat/model/StatItemTest.java
package uk.co.ianfield.devstat.model;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static com.google.common.truth.Truth.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

/**
 * Created by Ian on 20/11/2015.
 */
@RunWith(JUnit4.class)
public class StatItemTest {

    @Test
    public void testToString() {
        StatItem item = mock(StatItem.class);
        when(item.getTitle()).thenReturn("a");
        when(item.getInfo()).thenReturn("b");
        assertThat(item.getTitle()).isEqualTo("a");

        // Can't mock toString()
        item = new StatItem();
        item.setTitle("a");
        item.setInfo("b");
        assertThat(item.toString()).isEqualTo("a: b");
    }
}

File: app/src/test/java/uk/co/ianfield/devstat/model/StatItemTest.kt
package uk.co.ianfield.devstat.model

import com.google.common.truth.Truth.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4
import org.mockito.Mock
import org.mockito.MockitoAnnotations.initMocks
import org.mockito.Mockito.`when` as whenMock

/**
 * Created by Ian on 20/11/2015.
 */
@RunWith(JUnit4::class)
class StatItemTest {
    @Mock lateinit var statItem: StatItem

    @Before
    fun setUp() {
        initMocks(this)
    }

    @Test
    @Throws(Exception::class)
    fun testToString() {
        whenMock(statItem.title).thenReturn("a")
        whenMock(statItem.info).thenReturn("b")
        assertThat(statItem.title).isEqualTo("a")

        // Can't mock toString()
        statItem = StatItem()
        statItem.title = "a"
        statItem.info = "b"
        assertThat(statItem.toString()).isEqualTo("a: b")
    }
}
